src/rocksdb/src/lib.rs

```diff
@@ -244,48 +249,55 @@ where
         let dynamic_config = tuning_config.dynamic.clone();
         if options.cleanup_on_new && instance_path.exists() {
             let instance_path_owned = instance_path.to_owned();
-            let legacy_instance_path_owned = legacy_instance_path.to_owned();
-            mz_ore::task::spawn_blocking(
-                || {
-                    format!(
-                        "RocksDB instance at {}: cleanup on creation",
-                        instance_path.display()
+
+            // We require that cleanup of the DB succeeds. Otherwise, we could open a DB with old,
+            // incorrect data. Because of races with dataflow shutdown, we retry a few times here.
+            // 1s with a 2x backoff is ~30s after 5 tries.
```

Comment:
Can you add a TODO here that indicates this entire thing could and should go away when the controller can await the dataflow's destruction before restarting it?

Reply:
yes definitely!